src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -230,7 +230,18 @@
       <OldBuildVersionFiles Include="$(BuildVersionFilePath)BuildVersions-*.props" Exclude="$(BuildVersionFilePath)%(BuildVersionFileItem.Filename).props" />
     </ItemGroup>
     <Delete Files="@(OldBuildVersionFiles)" TreatErrorsAsWarnings="true"/>
+  </Target>
 
+  <Target Name="GenerateBuiltByString" Condition="'$(OSEnvironment)'!='Windows_NT' AND '$(BuiltByString)'==''">
+    <Exec Command="whoami" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true">
```

Comment:
Are these tools guaranteed to be on the machines? If they aren't then I would probably just avoid this completely.